sensors/trigger.go

```diff
@@ -213,15 +230,78 @@ func (sec *sensorExecutionCtx) executeTrigger(trigger v1alpha1.Trigger) error {
 		if err != nil {
 			return err
 		}
-		if err = sec.createResourceObject(trigger.Template, trigger.ResourceParameters, uObj); err != nil {
+		if err = sec.createResourceObject(&trigger, uObj); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// applyTriggerPolicy applies backoff and evaluates success/failure for a trigger
+func (sec *sensorExecutionCtx) applyTriggerPolicy(trigger *v1alpha1.Trigger, resourceInterface dynamic.ResourceInterface, name, namespace string) error {
+	err := wait.ExponentialBackoff(wait.Backoff{
+		Duration: trigger.Policy.Backoff.Duration,
+		Steps:    trigger.Policy.Backoff.Steps,
+		Factor:   trigger.Policy.Backoff.Factor,
+		Jitter:   trigger.Policy.Backoff.Jitter,
+	}, func() (bool, error) {
+		obj, err := resourceInterface.Get(name, metav1.GetOptions{})
+		if err != nil {
+			sec.log.WithError(err).WithField("resource-name", obj.GetName()).Error("failed to get triggered resource")
+			return false, nil
+		}
+
+		labels := obj.GetLabels()
+		if labels == nil {
```

Comment:
what about if the policy criteria is empty and i just care about obj creation?

Reply:
That is the default behavior. If you don't provide the policy, the sensor will just create the trigger object and move to next one if any